### ISSUE TITLE
Reduce `runsc state` call.

### ISSUE THREAD

**Issue by Random-Liu** [google/gvisor-containerd-shim#39]
We may want similar optimization with https://github.com/containerd/containerd/pull/3711.

Or leverage https://github.com/google/gvisor/issues/238

**Comment by ianlewis** [comment]
If the caching is done in containerd via containerd/containerd#3711 do we need to make any changes to the shim? Or is it just a matter if updating our dependencies?

**Comment by Random-Liu** [comment]
Yeah, we still need it, because we have corresponding code on our side
https://github.com/google/gvisor-containerd-shim/tree/master/pkg/v1/proc

**Comment by ianlewis** [comment]
I think caching the state is ok for now. Implementing a urpc interface to the sandbox is a bit more involved.
